lib/server.js

```diff
@@ -179,34 +179,30 @@ function handleResponse(opts, req, resp, response) {
         var body;
         // Prepare error responses for the client
         if (response.status >= 400) {
-            if (!response.body) {
-                response.body = {};
-            }
+            rh['content-type'] = rh['content-type']
+                || 'application/problem+json';
+
             // Whitelist fields to avoid leaking sensitive info
-            var rBody = response.body;
+            var rBody = response.body || {};
             body = {
                 type: rBody.type,
                 title: rBody.title,
-                method: rBody.method,
+                method: rBody.method || req.method,
                 detail: rBody.detail || rBody.description,
-                uri: rBody.uri
+                instance: rBody.instance || rBody.uri || req.uri
```

Comment:
Why wouldn't they? a `req` here is the root request, so the uri would be like `/en.wikipedia.org/v1/page/html/Test` which exactly points to the resource which the problem occurred for - so it's an instance of problem occurrence. I could rename it back to uri if you don't think my logic is what the RFC mean